cadc-util/src/main/java/ca/nrc/cadc/net/HttpPost.java

```diff
@@ -540,6 +553,26 @@ private void writeFilePart(String fieldName, File uploadFile, Writer w)
         }
         log.debug("file part length: " + len);
     }
+
+    private void writeFilePart(String fieldName, FileContent uploadContent, Writer w)
+        throws IOException
+    {
+        StringBuilder sb = new StringBuilder();
+        log.warn(uploadContent.getContentType());
+        sb.append(LINE_FEED).append("--" + MULTIPART_BOUNDARY);
+        // 'filename' for data entry is needed so this data is treated as
+        // stream input by the accepting web service
+        sb.append(LINE_FEED).append("Content-Disposition: form-data; name=\"" + fieldName + "\";"
+            + " filename=\"dummyFile\"");
+
+        sb.append(LINE_FEED).append("Content-Type: " + uploadContent.getContentType() + ";");
```

Comment:
This line should only be added if contentType is not null; it could otherwise fail in an unpredictable way.